cmd/gitserver/server/server.go

```diff
@@ -1140,6 +1142,96 @@ func matchCount(cm *protocol.CommitMatch) int {
 	return 1
 }
 
+func (s *Server) handleBatchLog(w http.ResponseWriter, r *http.Request) {
+	operations := s.ensureOperations()
+
+	// Run git log for a single repository.
+	// Invoked multiple times from the handler defined below.
+	performGitLogCommand := func(ctx context.Context, repoCommit api.RepoCommit, format string) (output string, isRepoCloned bool, err error) {
+		ctx, endObservation := operations.batchLogSingle.With(ctx, &err, observation.Args{
+			LogFields: []log.Field{
+				log.String("repoCommit.Repo", string(repoCommit.Repo)),
+				log.String("repoCommit.CommitID", string(repoCommit.CommitID)),
+				log.String("format", format),
+			},
+		})
+		defer func() {
+			endObservation(1, observation.Args{LogFields: []log.Field{
+				log.Bool("isRepoCloned", isRepoCloned),
+			}})
+		}()
+
+		dir := s.dir(repoCommit.Repo)
+		if !repoCloned(dir) {
+			return "", false, nil
+		}
+
+		var buf bytes.Buffer
+		cmd := exec.CommandContext(ctx, "git", "log", format, string(repoCommit.CommitID))
+		dir.Set(cmd)
+		cmd.Stdout = &buf
+
+		if _, err := runCommand(ctx, cmd); err != nil {
+			return "", true, err
+		}
+
+		return buf.String(), true, nil
+	}
+
+	// Handles the /batch-log route
+	instrumentedHandler := func(ctx context.Context) (statusCodeOnError int, err error) {
+		ctx, logger, endObservation := operations.batchLog.WithAndLogger(ctx, &err, observation.Args{})
+		defer func() {
+			endObservation(1, observation.Args{LogFields: []log.Field{
+				log.Int("statusCodeOnError", statusCodeOnError),
+			}})
+		}()
+
+		// Read request body
+		var req protocol.BatchLogRequest
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			return http.StatusBadRequest, err
+		}
+		logger.Log(log.Int("req.numRepoCommits", len(req.RepoCommits)))
+		logger.Log(log.String("req.format", req.Format))
+
+		// Perform requests in each repository in the input batch. We do this synchronously
+		// today so that this endpoint remains simple. If it is determined that performing
+		// the git log requests in parallel _on each shard_ is necesesary, we should keep
+		// the following sequence of actions as simple as possible.
+
+		results := make([]protocol.BatchLogResult, 0, len(req.RepoCommits))
+		for _, repoCommit := range req.RepoCommits {
+			output, ok, err := performGitLogCommand(ctx, repoCommit, req.Format)
+			if err == nil && !ok {
```

Comment:
```suggestion
			output, isRepoCloned, err := performGitLogCommand(ctx, repoCommit, req.Format)
			if err == nil && !isRepoCloned {
```
nit: Matching the names on the results in the `performGitLogCommand` with the bindings here makes it easier to independently understanding the body of `performGitLogCommand` and the usage of the `ok` variable here.